### PR TITLE
docs: Huge Docs revamp

### LINKED ISSUE #1054: [REQUEST] Update function parameters
### Describe the feature.

There are quite a few places where the function parameters don't quite line up with what it expects. This is mostly referencing `*args: Tuple[...]` and `**kwargs: Dict`, which when accessed inside of a function or when calling it, give the wrong typehints. In many docstrings, not only does the `*args`/`**kwargs` discrepancy show up, but on some IDEs (pycharm, possibly more), parameters can fail to appear when accessing the formatted docstring (often caused by `*args` or `parameter?`)

### Code of Conduct

- [X] I agree to follow the contribution requirements.

### PR DESCRIPTION
**Pull request by mAxYoLo01**
## About

This PR's goal is to massively revamp the docs to a more readable version.

In this PR is included:
- Rethink the general layout of the docs to make them more user-friendly, and highlight the important part
- Remove certain "private" files from being generated in docs (cache, error, gateway, ...) as they should not be public
- Hide private methods & functions from the docs
- Fix multiple docs generation warnings & errors
- Remove all `?` from docstrings
- Combining `:param:` and `:type:` into a single line in docstrings

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [ ] To add a new feature
  - [x] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [x] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

**Review by sillyElvy — CHANGES_REQUESTED** [review]
target branch. Also why the removal of error docs?

**Comment by mAxYoLo01** [comment]
Here is the new layout of the docs:
image
image

File: docs/api.http.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 HTTP
 ====
 
+.. warning::
```
Comment (sillyElvy) [inline docs/api.http.rst]:
```suggestion
.. important::
```
maybe?

**Comment by mAxYoLo01** [comment]
> target branch. Also why the removal of error docs?

I feel like since this is such a massive update, it belongs in unstable. Not only docs have been changed. Also, the docs branch is quite behind.

The error page had nothing important in it, kind of like the cache one, nothing that a random person needs to see.

**Comment by sillyElvy** [comment]
> > target branch. Also why the removal of error docs?
> 
> I feel like since this is such a massive update, it belongs in unstable. Not only docs have been changed. Also, the docs branch is quite behind.
> 
> The error page had nothing important in it, kind of like the cache one, nothing that a random person needs to see.

Alright, that is fair

and yeah I didn't see the code changes tbf

**Comment by mAxYoLo01** [comment]
I'm going to have to regenerate all the locale docs too with all the modifications, should I do it here or another PR?

File: docs/events.rst [hunk 2]
```diff
@@ -19,18 +19,18 @@ Generally, you can listen to an event like this:
 
     # possibility 1
     @bot.event
-    async def on_(...):
+    async def on_xxx(...):
```
Comment (Damego) [inline docs/events.rst:22]: Can you add this link https://discord.com/developers/docs/topics/gateway-events#receive-events

File: docs/events.rst [hunk 3]
```diff
@@ -74,7 +74,7 @@ as long as you don't absolutely need it.
 
 
 Event: ``on_start``
-^^^^^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Damego) [inline docs/events.rst:77]: Why you added more `^` symbols?

File: interactions/api/dispatch.py [hunk 4]
```diff
@@ -27,12 +27,9 @@ def dispatch(self, __name: str, *args, **kwargs) -> None:
...
         Dispatches an event given out by the gateway.
 
-        :param __name: The name of the event to dispatch.
-        :type __name: str
-        :param *args: Multiple arguments of the coroutine.
-        :type *args: list[Any]
-        :param **kwargs: Keyword-only arguments of the coroutine.
-        :type **kwargs: dict
+        :param str __name: The name of the event to dispatch.
+        :param list[Any] \*args: Multiple arguments of the coroutine.
```
Comment (Damego) [inline interactions/api/dispatch.py]: `*args` is an `Any` and `args` is a `Tuple[Any]`, not a list. Same thing for `kwargs`

File: docs/events.rst (hunk 3 above)
Comment (mAxYoLo01) [reply docs/events.rst:77]: Idk, I think I didn't understand yet the "rule" that it doesn't matter how many characters there are, as long as it's more than the length of the title

File: docs/events.rst (hunk 2 above)
Comment (mAxYoLo01) [reply docs/events.rst:22]: this link is already present in this page line 154

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Has http been removed from index and searching?

**Comment by mAxYoLo01** [comment]
> Has http been removed from index and searching?

Yes, as these methods should not be used by developers except if they have very good reasons to, which in this case I would guess they would check the source code at this point.

**Comment by sillyElvy** [comment]
> > Has http been removed from index and searching?
> 
> Yes, as these methods should not be used by developers except if they have very good reasons to, which in this case I would guess they would check the source code at this point.

Yeah I'm pro removing, sick of seeing "why is `interactions.api.http.guild.GuildRequest.get_member(id)` not working" xD

**Review by sillyElvy — APPROVED** [review]
I think this is fine and I don't want to make you a hell conflict when I make another refactor, or the other way around

**Pull request merged**